scripts/Synapse/convertParquetMPI.ipynb

```diff
@@ -1,152 +1,171 @@
+    "client_id = client.get_secret(\"synapse-client-id\").value\n",
+    "client_secret = client.get_secret(\"synapse-client-secret\").value\n",
+    "\n",
+    "# Get access token for record linkage container app\n",
+    "tenant_id = \"$TENANT_ID\"\n",
+    "url = f\"https://login.microsoftonline.com/{tenant_id}/oauth2/v2.0/token\"\n",
+    "\n",
+    "data = {\n",
+    "    'grant_type': 'client_credentials',\n",
+    "    'client_id': client_id,\n",
+    "    'client_secret': client_secret,\n",
+    "    'scope': 'api://phdi-dev-record-linkage/.default'\n",
+    "}\n",
+    "\n",
+    "response = requests.post(url, data=data)\n",
+    "access_token = response.json()['access_token']\n",
+    "\n",
+    "# Make request to record linkage container app\n",
+    "record_linkage_url = client.get_secret(\"record-linkage-url\") + \"/link-record\"\n",
+    "ingestion_url = client.get_secret(\"ingestion-url\")\n",
+    "std_names_url = ingestion_url + \"/standardize_names\"\n",
+    "std_phones_url = ingestion_url + \"/standardize_phones\"\n",
+    "std_address_url = ingestion_url + \"/geocode_bundle\"\n",
+    "headers = {\n",
+    "    'Authorization': f'Bearer {access_token}'\n",
+    "}\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Read the MPI parquet data into a spark dataframe. Iterate over each row of patient data in the dataframe and convert to a FHIR bundle and associated iris_id. Create a POST request to the record linkage container with FHIR bundle and iris_id."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Convert and post data from mpi_incoming_file_path\n",
+    "def convert(mpi_incoming_file_path):\n",
+    "    df = spark.read.parquet(mpi_incoming_file_path)\n",
+    "    for row in df.collect():\n",
+    "        iris_id, fhir_bundle = convert_to_patient_fhir_resources(row.asDict())\n",
+    "\n",
+    "        # Add metadata to LAC Extract denoting it came from IRIS\n",
+    "        # We know this URI is ~probably~ not right but Brady might know what to use...\n",
+    "        fhir_bundle[\"meta\"] = {\"source\": \"uri:iris\"}\n",
+    "\n",
+    "        data = {\n",
+    "            'bundle': fhir_bundle,\n",
+    "            'external_person_id': iris_id\n",
+    "        }\n",
+    "        \n",
+    "        # Perform pipeline standardization on extracted data to mirror how\n",
+    "        # non-extracted data gets handled by the pipeline\n",
+    "        resp = requests.post(std_names_url, headers=headers, json=data)\n",
+    "        data['bundle'] = resp.get(\"bundle\")\n",
+    "        resp = requests.post(std_phones_url, headers=headers, json=data)\n",
+    "        data['bundle'] = resp.get(\"bundle\")\n",
+    "        resp = requests.post(std_address_url, headers=headers, json=data)\n",
+    "        data['bundle'] = resp.get(\"bundle\")\n",
```

Comment:
We should standardize DOB here as well because it is part of the pipeline now.

Reply:
yep good call, will add that in